kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -166,7 +174,8 @@ public class TableConfig<T> {
           "none", /* default values */
           (engineOpt, v) -> ColumnMappingMode.fromTableConfig(v),
           value -> true,
-          "Needs to be one of none, id, name.");
+          "Needs to be one of none, id, name.",
+          true);
```

Comment:
can you add the property: `delta.columnMapping.maxColumnId` which is not editable and a negative test to make sure it is not editable.